src/ir/item.rs

```diff
@@ -825,6 +826,50 @@ impl Item {
             _ => None,
         }
     }
+
+    /// Returns whether the item is a constified module enum
+    fn is_constified_enum_module(&self, ctx: &BindgenContext) -> bool {
```

Comment:
I don't think this is about the number of layers to resolve. I think the logic in `is_constified_enum_module` is wrong, and the "alias to resolved type ref to alias" logic is just masking it.